core/block/object/objectcreator/migration.go

```diff
@@ -0,0 +1,25 @@
+package objectcreator
+
+import (
+	"context"
+
+	"github.com/anyproto/anytype-heart/core/migration"
+	"github.com/anyproto/anytype-heart/space/clientspace"
+)
+
+func (s *service) RunMigrations(ctx context.Context, space clientspace.Space) {
```

Comment:
I think you can have a single context for all RunMigration calls here. Because you still have a single Close. RunMigrations now is not thread-safe (because of appending to the slice). Just create the ctx and cancel in service Init.

Reply:
Thank you! I forgot that appending to slice is not thread-safe, but your solution requires storing context inside ObjectCreator structure, that is a bad practice. Moreover, we pass contexts from space builder, so we have context for each space.

Maybe it is better to modify CancelFunc slice under lock?

Reply:
However, use of single ctx is pretty accurate. I did it acc to your suggestion